xmtp_api_grpc/src/replication_client.rs

```diff
@@ -462,3 +443,83 @@ impl XmtpTestClient for ClientV4 {
         todo!()
     }
 }
+impl ClientV4 {
+    #[tracing::instrument(level = "trace", skip_all)]
+    async fn query_v4_envelopes(
+        &self,
+        topics: Vec<Vec<u8>>,
+    ) -> Result<Vec<Vec<OriginatorEnvelope>>, Error> {
+        let requests = topics.iter().map(|topic| async {
+            let client = &mut self.client.clone();
+            let v4_envelopes = client
+                .query_envelopes(QueryEnvelopesRequest {
+                    query: Some(EnvelopesQuery {
+                        topics: vec![topic.clone()],
+                        originator_node_ids: vec![],
+                        last_seen: None,
+                    }),
+                    limit: 100,
```

Comment:
To avoid bugs, I think we should take in the `last_seen` and `limit` from the caller of this method, and honor whatever semantics were previously used